File: frontend/src/types/report.ts
export interface GstSummary {
  collected: number;
  paid: number;
  net: number;
}

export interface PaygSummary {
  total_gross: number;
  total_payg: number;
  total_super: number;
  employee_count: number;
}

export interface RdSpendByProject {
  project_name: string;
  amount: number;
}

export interface UnresolvedException {
  type: string;
  count: number;
}

export interface MonthlyReport {
  month: number;
  year: number;
  gst_summary: GstSummary;
  payg_summary: PaygSummary;
  total_operating_spend: number;
  rd_eligible_spend: number;
  rd_spend_by_project: RdSpendByProject[];
  outstanding_issues_count: number;
  missing_evidence_count: number;
  unresolved_exceptions: UnresolvedException[];
}

export interface RdProjectSpend {
  project_id: string;
  project_name: string;
  spend: number;
  percentage: number;
}

export interface RdSummary {
  financial_year: string;
  total_rd_spend: number;
  by_project: RdProjectSpend[];
  by_category: Record<string, number>;
}

export interface BasStatusInfo {
  current_period: string | null;
  status: string;
}

export interface PaygStatusInfo {
  current_month: string | null;
  status: string;
}

export interface ComplianceStatus {
  bas_status: BasStatusInfo;
  payg_status: PaygStatusInfo;
  evidence_gaps_count: number;
  unresolved_exceptions_count: number;
  overall_status: 'good' | 'warning' | 'critical';
}

File: backend/app/schemas/report.py
from decimal import Decimal
from typing import List, Dict, Optional
from pydantic import BaseModel


class GstSummary(BaseModel):
    collected: Decimal
    paid: Decimal
    net: Decimal


class PaygSummary(BaseModel):
    total_gross: Decimal
    total_payg: Decimal
    total_super: Decimal
    employee_count: int


class RdSpendByProject(BaseModel):
    project_name: str
    amount: Decimal


class UnresolvedException(BaseModel):
    type: str
    count: int


class MonthlyReportResponse(BaseModel):
    month: int
    year: int
    gst_summary: GstSummary
    payg_summary: PaygSummary
    total_operating_spend: Decimal
    rd_eligible_spend: Decimal
    rd_spend_by_project: List[RdSpendByProject]
    outstanding_issues_count: int
    missing_evidence_count: int
    unresolved_exceptions: List[UnresolvedException]


class RdProjectSpend(BaseModel):
    project_id: str
    project_name: str
    spend: Decimal
    percentage: Decimal


class RdSummaryResponse(BaseModel):
    financial_year: str
    total_rd_spend: Decimal
    by_project: List[RdProjectSpend]
    by_category: Dict[str, Decimal]


class BasStatusInfo(BaseModel):
    current_period: Optional[str]
    status: str


class PaygStatusInfo(BaseModel):
    current_month: Optional[str]
    status: str


class ComplianceStatusResponse(BaseModel):
    bas_status: BasStatusInfo
    payg_status: PaygStatusInfo
    evidence_gaps_count: int
    unresolved_exceptions_count: int
    overall_status: str  # good/warning/critical


class MonthlyReportExportRow(BaseModel):
    category: str
    item: str
    amount: Decimal
    details: Optional[str] = None
